### PR TITLE
feat: support forked repo

### PR DESCRIPTION
**Pull request by lijie0123**
Why:
- to support forked repo
- for enterprise-plugin TIDB_HASH can accept repo name before sha1 to accomplish plugin with forked repo

### CODE REVIEW

File: jenkins/pipelines/cd/atom-jobs/build-common.groovy [hunk 1]
```diff
@@ -352,8 +366,9 @@ def checkoutCode() {
                                         [$class: 'CleanBeforeCheckout']], submoduleCfg: [],
                             userRemoteConfigs: [[credentialsId: 'github-sre-bot-ssh',
                                                 refspec      : specRef,
-                                                url          : 'git@github.com:pingcap/tidb.git']]]
+                                                url          : tidb_repo]]]
             }
+            sh "git reset --hard ${TIDB_HASH}"
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/atom-jobs/build-common.groovy:371]: why need another git cli step to reset?

**Review by purelind — COMMENTED** [review]
1. I think the current pipeline is already too complex and continuing to add new parameters 
which would make people confused may not be a good choice.
2. Support fork repo is a necessary feature or not, the current go version depends on tag or branch to determine, support fork reop build does not guarantee the use of normal go version

**Comment by lijie0123** [comment]
1. It is pain to add more logic to this big script, but  the customer need forked repo because the cloud use forked repo , and it is better to be online before March
2. go version is selected by version prior than branch

**Comment by purelind** [comment]
> 1. It is pain to add more logic to this big script, but  the customer need forked repo because the cloud use forked repo , and it is better to be online before March
> 2. go version is selected by version prior than branch, and version must be given

OK, LGTM, it will take some time to refactor the build-common.

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**